x-pack/plugin/inference/src/yamlRestTest/java/org/elasticsearch/xpack/inference/InferenceRestIT.java

```diff
@@ -21,9 +21,8 @@ public class InferenceRestIT extends ESClientYamlSuiteTestCase {
     public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
         .setting("xpack.security.enabled", "false")
         .setting("xpack.security.http.ssl.enabled", "false")
-        .plugin("x-pack-inference")
         .plugin("inference-service-test")
-        .distribution(DistributionType.INTEG_TEST)
+        .distribution(DistributionType.DEFAULT)
```

Comment:
I had to change distrib to default, as I couldn't add the ML plugin otherwise (needed for testing the `text_expansion` query against the indexed results, and prove that they can be queried using a standard query)

Reply:
Why couldn't the ML module be added? Did you encounter a particular error?

Reply:
I double what mark is asking, is adding the ml module for the tests breaking things?

Reply:
The existance of the ML module itself couldn't be the issue, since the default distribution already includes it. Perhaps there was just an issue in setup/configuration here? I think we should try and sort this out so we can retain the benefits of using the integ test distribution.

Reply:
Thanks for looking into this. This is what I tried:

on InferenceRestIT:

```java
    @ClassRule
    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
        .setting("xpack.security.enabled", "false")
        .setting("xpack.security.http.ssl.enabled", "false")
        .plugin("x-pack-inference")
        .plugin("x-pack-ml")
        .plugin("inference-service-test")
        .distribution(DistributionType.INTEG_TEST)
        .build();
```

on build.gradle:

```groovy
dependencies {
  implementation project(path: ':libs:elasticsearch-logging')
  compileOnly project(":server")
  compileOnly project(path: xpackModule('core'))
  testImplementation(testArtifact(project(xpackModule('core'))))
  testImplementation project(':modules:reindex')
  clusterPlugins project(':x-pack:plugin:inference:qa:test-service-plugin')
  clusterPlugins project(':x-pack:plugin:ml')

  api "com.ibm.icu:icu4j:${versions.icu4j}"
}
```

This makes test cluster orchestration to fail with `IllegalStateException: plugins can not have native controllers`:

```
[2024-05-06T14:32:02,773][INFO ][o.e.t.c.l.AbstractLocalClusterFactory] [[test-cluster-node-executor-0]] Distribution for node 'test-cluster-0': DefaultDistributionDescriptor{version=8.15.0, snapshot=true, distributionDir=/Users/cdelgado/dev/workspace/worktree/feature/semantic-text/distribution/archives/integ-test-zip/build/install, type=INTEG_TEST}
[2024-05-06T14:32:02,845][INFO ][o.e.t.c.l.AbstractLocalClusterFactory] [[test-cluster-node-executor-0]] Installing plugins [x-pack-ml, x-pack-inference, inference-service-test] into node 'test-cluster-0
[2024-05-06T14:32:03,732][INFO ][process-output           ] [[elasticsearch-plugin-log-forwarder]] -> Installing file:///Users/cdelgado/dev/workspace/worktree/feature/semantic-text/x-pack/plugin/ml/build/distributions/x-pack-ml-8.15.0-SNAPSHOT.zip
[2024-05-06T14:32:03,732][INFO ][process-output           ] [[elasticsearch-plugin-log-forwarder]] -> Downloading file:///Users/cdelgado/dev/workspace/worktree/feature/semantic-text/x-pack/plugin/ml/build/distributions/x-pack-ml-8.15.0-SNAPSHOT.zip
[2024-05-06T14:32:10,584][INFO ][process-output           ] [[elasticsearch-plugin-log-forwarder]] -> Failed installing file:///Users/cdelgado/dev/workspace/worktree/feature/semantic-text/x-pack/plugin/ml/build/distributions/x-pack-ml-8.15.0-SNAPSHOT.zip
[2024-05-06T14:32:10,584][INFO ][process-output           ] [[elasticsearch-plugin-log-forwarder]] -> Rolling back file:///Users/cdelgado/dev/workspace/worktree/feature/semantic-text/x-pack/plugin/ml/build/distributions/x-pack-ml-8.15.0-SNAPSHOT.zip
[2024-05-06T14:32:10,648][INFO ][process-output           ] [[elasticsearch-plugin-log-forwarder]] -> Rolled back file:///Users/cdelgado/dev/workspace/worktree/feature/semantic-text/x-pack/plugin/ml/build/distributions/x-pack-ml-8.15.0-SNAPSHOT.zip
[2024-05-06T14:32:10,648][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] Exception in thread "main" java.lang.IllegalStateException: plugins can not have native controllers
[2024-05-06T14:32:10,648][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.plugins.cli.InstallPluginAction.loadPluginInfo(InstallPluginAction.java:865)
[2024-05-06T14:32:10,648][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.plugins.cli.InstallPluginAction.installPlugin(InstallPluginAction.java:923)
[2024-05-06T14:32:10,648][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.plugins.cli.InstallPluginAction.execute(InstallPluginAction.java:254)
[2024-05-06T14:32:10,648][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.plugins.cli.InstallPluginCommand.execute(InstallPluginCommand.java:89)
[2024-05-06T14:32:10,648][ERROR][process-output           ] [[elasticsearch-plugin-log-forwarder]] 	at org.elasticsearch.common.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:54)
```

I was unable to find any other example of an integration test that uses the ML plugin as a dependency. 

What am I missing?

What am I missing?

Reply:
@carlosdelest please set the following node setting on your integration test nodes:

```
        builder.put(MachineLearningField.AUTODETECT_PROCESS.getKey(), false);
```

Reply:
```
.setting("xpack.ml.autodetect_process", "false")
```

Reply:
Still no luck, same error with:

```java
    @ClassRule
    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
        .setting("xpack.security.enabled", "false")
        .setting("xpack.security.http.ssl.enabled", "false")
        .setting("xpack.ml.autodetect_process", "false")
        .plugin("x-pack-inference")
        .plugin("x-pack-ml")
        .plugin("inference-service-test")
        .distribution(DistributionType.INTEG_TEST)
        .build();
```

```groovy
dependencies {
  implementation project(path: ':libs:elasticsearch-logging')
  compileOnly project(":server")
  compileOnly project(path: xpackModule('core'))
  testImplementation(testArtifact(project(xpackModule('core'))))
  testImplementation project(':modules:reindex')
  clusterPlugins project(':x-pack:plugin:inference:qa:test-service-plugin')
  clusterPlugins project(':x-pack:plugin:ml')

  api "com.ibm.icu:icu4j:${versions.icu4j}"
}
```

I see some IntegTests using this setting, but I haven't been able to find how that is used with YAML tests 😢

Reply:
@mark-vieira , @benwtrent , I'd appreciate your help here for using the `DistributionType.INTEG_TEST` - I've been unable to do so.